pkg/kubelet/status/generate.go

```diff
@@ -62,11 +64,12 @@ func GenerateContainersReadyCondition(spec *v1.PodSpec, containerStatuses []v1.C
 
 	// If all containers are known and succeeded, just return PodCompleted.
 	if podPhase == v1.PodSucceeded && len(unknownContainers) == 0 {
-		return v1.PodCondition{
-			Type:   v1.ContainersReady,
-			Status: v1.ConditionFalse,
-			Reason: PodCompleted,
-		}
+		return generateContainersReadyConditionForTerminalPhase(podPhase)
```

Comment:
Going back and looking through this, the `len(unknownContainers) == 0` above has me worried.

There are two phases in play - the apiserver phase, and the one the Kubelet tracks.  Setting phase to terminal on the apiserver is roughly the same outcome as deleting the pod - the kubelet should inexorably converge the actual pod state to stopped.  Likewise if the Kubelet observes a failure or evicts the pod, the phase should inexorably converge.  However, internal to the Kubelet status loop the phase should *match* the actual observed state of the pod containers, and so we should never be in a terminal state here without these containers being in some state we recognize (I think).

This is something I want to have laid out in the pod lifecycle clarification KEP so we can look at all the factors and eliminate my "shoulds" from above - it might be that the correct status when succeeded + > 1 unknown container = pod ready = false here, always.  But it's super subtle, and we need this sort of stuff in a clear top level doc, not in comment threads in a PR.   I'll add it to the list.